### PR TITLE
Hardware and configuration selections and travis build server setup

### PR DESCRIPTION
**Pull request by unitware**
(This PR is admittely a bit chunky, please let me know if you want me to split it more commits)

I have added support to build all combinations of the application on travis (see https://travis-ci.org/unitware/QuadMeUp_Crossbow).

The matrix is set up with two hardwares (feather 32u4, feather M0) and some different sets of defines testing both rx and tx, and variants of tx to use ppm input or sbus input.

All combinations are specified in configurations/config_nn_nn.h and one of these files are moved to replace the config.h file when building. (I did it this way since I did not find a way to inject the configurations when scripting the build.)

There are also some changes to the tx input implementation to unify the interface between ppm and sbus input. I hope I got the licensing right in that case (combining MPL and GPL3)

### CODE REVIEW

**Comment by DzikuVx** [comment]
@unitware could you split this into 2 separate PRs please? 
Besides that, cool, thanks!

File: crossbow/ppm_reader.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
```
Comment (DzikuVx) [inline crossbow/ppm_reader.cpp]: Latest version is not using PPM decoding and probably will never be. OpenTX 2.2.1 already has been published and there is no need to fallback to such old protocol as PPM

**Review by DzikuVx — CHANGES_REQUESTED** [review]
Split Travis into seprate PR please and remove ppm decoding since it's not used and should not be used

**Comment by unitware** [comment]
I can split the PR and leave out the ppm, no problem. But I’d like yo keep the infrastructure to be able to select tx input. I do not have any hardware that outputs sbus so I need to keep those files for my own use. 

The GPL3 will disappear when removing the ppm_reader and that makes it easier :)

Skickat från min iPhone

**Comment by DzikuVx** [comment]
Yea... I keep forgetting that PPMReader is GPL not MPL. Even my own code ;)

**Comment by unitware** [comment]
so, I removed the GPL3 stuff - that is the ppm_reader and the licensing text.

I left the TxInput base class to make it easier for me to keep in track with your master later on. I am using it for ppm and maybe later also a variant with analog inputs so I can build a new tx from an old one with just the feather board inside. 

Does this feel ok?

**Comment by DzikuVx** [comment]
I will gry to test in a next few days on a real hardware of mine

**Comment by DzikuVx** [comment]
OK, let's see how it goes... compiles on Lora32u4 had no time to test it further tho

**Comment by unitware** [comment]
great, now log in to travis and enable the repo :)
https://travis-ci.org/DzikuVx

**Pull request merged**